aderyn_core/src/detect/high/arithmetic_underflow_overflow.rs

```diff
@@ -0,0 +1,162 @@
+            version_req_below_0_8(&req).then(|| {
+                context
+                    .assignments()
+                    .iter()
+                    .filter(|assignment| assignment.operator == "+=" || assignment.operator == "-=")
+                    .for_each(|assignment| capture!(self, context, assignment));
+                context
+                    .binary_operations()
+                    .iter()
+                    .filter(|binary_op| binary_op.operator == "+" || binary_op.operator == "-")
+                    .for_each(|op| capture!(self, context, op));
+            });
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn title(&self) -> String {
+        String::from("`SafeMath` library should be imported and used to prevent underflow and overflow in solidity compilers below `0.8.0`")
+    }
+
+    fn description(&self) -> String {
+        String::from("Use `SafeMath` library for arithmetic operations to avoid underflow and/or overflow in solidity compilers below `0.8.0`")
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::High
+    }
+
+    fn instances(&self) -> BTreeMap<(String, usize, String), NodeID> {
+        self.found_instances.clone()
+    }
+
+    fn name(&self) -> String {
+        format!("{}", IssueDetectorNamePool::ArithmeticUnderflowOverflow)
+    }
+}
+
+#[cfg(test)]
+mod arithmetic_underflow_overflow_tests {
+    use crate::detect::detector::{detector_test_helpers::load_contract, IssueDetector};
+
+    use super::ArithmeticUnderflowOverflowDetector;
+
+    #[test]
+    fn test_arithmetic_underflow_overflow() {
+        let context = load_contract(
+            "../tests/contract-playground/out/ArithmeticUnderflowOverflow.sol/ArithmeticUnderflowOverflow.json",
+        );
+
+        let mut detector = ArithmeticUnderflowOverflowDetector::default();
+        // assert that the detector finds the public Function
+        let found = detector.detect(&context).unwrap();
+        assert!(found);
+        // assert that the detector returns the correct number of instances
+        assert_eq!(detector.instances().len(), 4);
+        // assert that the detector returns the correct severity
+        assert_eq!(
+            detector.severity(),
+            crate::detect::detector::IssueSeverity::High
+        );
+        // assert that the detector returns the correct title
+        assert_eq!(
+            detector.title(),
+            String::from("`SafeMath` library should be imported and used to prevent underflow and overflow in solidity compilers below `0.8.0`")
+        );
+        // assert that the detector returns the correct description
+        assert_eq!(detector.description(), String::from("Use `SafeMath` library for arithmetic operations to avoid underflow and/or overflow in solidity compilers below `0.8.0`"));
+    }
+
+    #[test]
+    fn test_arithmetic_underflow_overflow_constants_file() {
```

Comment:
@gr4yha7 I added this test to check a file that shouldn't capture any instances, but it does. (Please pull this branch)

It looks like your detector may still be catching contracts that have a pragma >=0.8.0.